configuration/src/gas/mod.rs

```diff
@@ -1,7 +1,21 @@
 //! Per-chain gas configurations
 
+use serde::{Deserialize, Serialize};
+
+mod defaults;
+
+/// Gas config types
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "type", content = "config", rename_all = "camelCase")]
```

Comment:
we've been moving away from type tagging and towards `serde(untagged)` enums generally because of the overhead it puts on the 

if we want it to specified in the config.json as follows:
```
"rinkeby": "evmDefault"
```

we should make it untagged, then make our serde implementation look like 

    ```
    fn deserialize<D>(d: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        let s = d.deserialize_string()?;
        if &s == "evmDefault" {
            Ok(Self(EVM_DEFAULT))
        } else { 
            Err(D::Error::custom(...))
        }
    }
    ```

Reply:
that'll give us flexibility for later defaults too

Reply:
a higher-level point is that the abstraction between custom EVM gas config and default EVM doesn't belong in this enum. This enum should abstract across EVM and non-EVM gas configs. we should probably push the defaulting down to the implementation of deserailize on `NomadGasConfig` ?

Reply:
Isn't `NomadGasConfig` VM-agnostic though? As long as our contract/module interfaces remain same, don't think there's anything VM-specific here (except maybe ethhelper). The idea was that other vm defaults would be in `NomadGasConfig[s]` enum

Reply:
gas config includes things like process gas which may not really make sense on other chains, depending on their contract-calling semantics. The real problem is that we don't know what they'll be yet 😬

Reply:
oh nevermind i'm off base here

Reply:
So do we want config to look something like this?
```
"gas": {
   "ethereum": {
      "vm": "ethereum",
      "config": "default",
   }
}
```

Reply:
honestly I think the look here is 
```
"gas": {
   "ethereum": "evmDefault"
}
```

Reply:
@ltchang2019 what about setting `#[serde(default = "something")]` for each field in the gas config struct? this would let us avoid introducing the second struct and enum variant at all 🤔

Reply:
alternately, a `FromStr` impl on the gas config struct, that handles the `"evmDefault"` and other 1 string cases, and is used by the `serde::Deserialize` implementation

Reply:
K addressed :)